pkg/sql/catalog/descpb/privilege_test.go

```diff
@@ -234,34 +234,116 @@ func TestPrivilegeValidate(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	id := ID(keys.MinUserDescID)
 	descriptor := NewDefaultPrivilegeDescriptor(security.AdminRole)
-	if err := descriptor.Validate(id); err != nil {
+	if err := descriptor.Validate(id, privilege.Table); err != nil {
 		t.Fatal(err)
 	}
 	descriptor.Grant("foo", privilege.List{privilege.ALL})
-	if err := descriptor.Validate(id); err != nil {
+	if err := descriptor.Validate(id, privilege.Table); err != nil {
 		t.Fatal(err)
 	}
 	descriptor.Grant(security.RootUser, privilege.List{privilege.SELECT})
-	if err := descriptor.Validate(id); err != nil {
+	if err := descriptor.Validate(id, privilege.Table); err != nil {
 		t.Fatal(err)
 	}
-	descriptor.Revoke(
-		security.RootUser, privilege.List{privilege.SELECT}, privilege.Table)
-	if err := descriptor.Validate(id); err == nil {
+	descriptor.Revoke(security.RootUser, privilege.List{privilege.SELECT},
+		privilege.Table)
+	if err := descriptor.Validate(id, privilege.Table); err == nil {
 		t.Fatal("unexpected success")
 	}
 	// TODO(marc): validate fails here because we do not aggregate
 	// privileges into ALL when all are set.
 	descriptor.Grant(security.RootUser, privilege.List{privilege.SELECT})
-	if err := descriptor.Validate(id); err == nil {
+	if err := descriptor.Validate(id, privilege.Table); err == nil {
 		t.Fatal("unexpected success")
 	}
 	descriptor.Revoke(security.RootUser, privilege.List{privilege.ALL}, privilege.Table)
-	if err := descriptor.Validate(id); err == nil {
+	if err := descriptor.Validate(id, privilege.Table); err == nil {
 		t.Fatal("unexpected success")
 	}
 }
 
+func TestTableDBPrivilegeValidate(t *testing.T) {
```

Comment:
I feel like these tests could be pulled into a single "table" style test that has each privilege type, and then the set of allowed and disallowed privileges

Reply:
Changed to something similar to this.